src/WebApi/Extensions/EndpointRouteBuilderExt.cs

```diff
@@ -2,21 +2,43 @@
 
 public static class EndpointRouteBuilderExt
 {
-    public static RouteHandlerBuilder ProducesGet<T>(this RouteHandlerBuilder builder) => builder
+    /// <summary>
+    /// Used for GET endpoints that return a single entity.
+    /// </summary>
+    public static RouteHandlerBuilder ProducesGetSingle<T>(this RouteHandlerBuilder builder) => builder
         .Produces<T>(StatusCodes.Status200OK)
+        .Produces(StatusCodes.Status404NotFound)
         .ProducesProblem(StatusCodes.Status500InternalServerError);
 
+    /// <summary>
+    /// Used for GET endpoints that return a list of entities.
+    /// </summary>
+    public static RouteHandlerBuilder ProducesGetList<T>(this RouteHandlerBuilder builder) => builder
```

Comment:
we're not really guarding against returning a single object.

Perhaps if we use `.Produces<T[]>(....)` or even `.Produces<IEnumerable<T>>(....)` we'd always indicate that an array is returned or

Reply:
This would also affect the users's declaration to be `.ProducesGetList<TodoItem>(....)` instead of `.ProducesGetList<TodoItem[]>(....)`

should we read this as a list of `TodoItem` or a list of `TodoItem[]` ?

Reply:
I don't think this would work.  The generic type needs to specify the `T` we can't overload the function based on `T` or `T[]`

![image](https://github.com/SSWConsulting/SSW.CleanArchitecture/assets/2636640/7ca07cf2-08eb-4bd6-9d26-ffeab1d5aca8)

![image](https://github.com/SSWConsulting/SSW.CleanArchitecture/assets/2636640/589a2cb4-9d48-4a10-bdc7-99509e96a075)

Reply:
Also, we shouldn't guard against returning single object as well - think of pagination 😀

I'd rather keep a generic `T` instead of specific "Single" or "GetList" and then let the user decide the return type

Reply:
The problem with returning a generic `T` is that some gets will return a 404 and some will return an empty 200 with no content.  However, the metadata for the API doesn't help the caller to determine which API's return a 404 and which don't.

The idea behind these changes is to try to distinguish between between which API's are looking up entities based on route parameters (and returning a 404) and which are more like a 'search' endpoint that may or may not return data, but will speciically NOT return a 404.

Reply:
interesting debate here :) @christoment and @danielmackay - lets pair up on this one and work through the use cases together

Reply:
After a conversation with @christoment today we decided it made sense for a GET API to always advertise a possible 404 status code even if this isn't possible from the code PoV.  This simplifies the problem above.